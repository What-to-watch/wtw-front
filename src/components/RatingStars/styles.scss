.rating-stars {
    display: flex;
    & label {
        cursor: pointer;
        display: flex;
        position: relative;
        width: 40px;
        height: 28px;
        & .rating-stars__star {
            padding: 0 7px;
            position: absolute;
            width: 100%;
            height: 100%;
            &.opacity {
                opacity: 0.5;
            }
        };
    }
    & input[type="radio"] {
        visibility: hidden;
    }
}