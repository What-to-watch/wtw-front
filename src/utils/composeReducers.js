import { reduce } from "ramda"

export const composeReducers = (...reducers) => (state,action) => {
    return reduce((accState,nextReducer) => nextReducer(accState,action) , state)(reducers)
}