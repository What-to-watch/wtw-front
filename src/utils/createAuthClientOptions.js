export const createAuthClientOptions = (token) => {
    return {
        headers: {
            authorization: `bearer ${token}`
        }
    }
}