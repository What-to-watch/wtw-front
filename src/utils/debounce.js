export default function debounce(fn,timer){
    let id = null;
    return (...args) => {
        if( id !== null ){
            clearTimeout(id)
        }
        id = setTimeout(() => fn(...args),timer);
    }
}