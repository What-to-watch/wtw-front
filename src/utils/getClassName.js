const getClassName = (obj) => Object.entries(obj).map(([className,pred]) => {
    const shouldConcat = typeof(pred) === "function" ? pred() : pred;
    return shouldConcat ? className : undefined
}).filter(Boolean).join(" ")

export default getClassName