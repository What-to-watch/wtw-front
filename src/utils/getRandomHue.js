export default function getRandomHue() {
    const min = Math.ceil(0);
    const max = Math.floor(360);
    return Math.floor(Math.random() * (max - min + 1)) + min;
}