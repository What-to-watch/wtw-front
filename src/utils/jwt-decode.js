import { Try } from "@juan-utils/ramda-structures";

const parse = x => JSON.parse(atob(x))

const jwtDecode = (token) => {
    const [ header, payload, verify] = token.split(".");
    return Try.from(() => ({
        header: parse(header),
        payload: parse(payload),
        verify,
    }))
}

export default jwtDecode;