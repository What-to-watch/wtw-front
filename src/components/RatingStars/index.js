import React, {useState} from 'react';
import getClassName from '../../utils/getClassName';

import './styles.scss';

const RatingStars = (props) => {
    const [ hover, setHover ] = useState(null);
    const { onChange, rating } = props

    const handleRating = (number) => () => {
        onChange(number);
    }

    return (
        <div className="rating-stars">
            {[...Array(5)].map((star, i) => {
                const ratingValue = i + 1;
                const classes = getClassName({
                    'rating-stars__star': true,
                    'opacity': hover && ratingValue > hover && ratingValue <= rating,
                })
                return (
                    <label key={`star${i}`}>
                        <input 
                            type="radio"
                            name="rating"
                            value={ratingValue}
                            onClick={handleRating(ratingValue)}
                        />
                        <img 
                            className={classes}
                            src={ ratingValue <= hover || ratingValue <= rating ? 'icons/fullStar.svg' : 'icons/Star.svg' }
                            alt="star"
                            onMouseEnter={() => setHover(ratingValue)}
                            onMouseLeave={() => setHover(null)}
                        />
                        
                    </label>
                )
            })}
        </div>
    )
}

export default RatingStars;