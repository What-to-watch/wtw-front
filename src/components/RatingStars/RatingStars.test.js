import React from 'react';
import RatingStars from './index';
import { render, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Spy } from '../../utils/testing'

describe ('RatingStars render', ()=>{
    let props = {
        rating: 4,
        onChange: x=>x,
    }
    let component = renderer.create(
        (<RatingStars {...props}/>),
    );

    let ratingStars = component.toJSON();

    it('RatingStars render', ()=>{

        const labels = component.root.findAllByType('label');

        expect(labels).toHaveLength(5);
        expect(ratingStars).toMatchSnapshot();

    });

    it('RatingStars onChange is executed with label', ()=>{
        const spy = Spy();

        props = { ...props, onChange: spy }

        render(<RatingStars {...props} />);

        fireEvent.click(screen.getAllByRole('radio')[0]);

        expect(spy.called).toBeTruthy();
        expect(spy.callCount).toBe(1);
        expect(spy.firstCall.args[0]).toBe(1);
    });

})