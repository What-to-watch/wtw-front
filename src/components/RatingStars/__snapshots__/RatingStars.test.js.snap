// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStars render RatingStars render 1`] = `
<div
  className="rating-stars"
>
  <label>
    <input
      name="rating"
      onClick={[Function]}
      type="radio"
      value={1}
    />
    <img
      alt="star"
      className="rating-stars__star"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="icons/fullStar.svg"
    />
  </label>
  <label>
    <input
      name="rating"
      onClick={[Function]}
      type="radio"
      value={2}
    />
    <img
      alt="star"
      className="rating-stars__star"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="icons/fullStar.svg"
    />
  </label>
  <label>
    <input
      name="rating"
      onClick={[Function]}
      type="radio"
      value={3}
    />
    <img
      alt="star"
      className="rating-stars__star"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="icons/fullStar.svg"
    />
  </label>
  <label>
    <input
      name="rating"
      onClick={[Function]}
      type="radio"
      value={4}
    />
    <img
      alt="star"
      className="rating-stars__star"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="icons/fullStar.svg"
    />
  </label>
  <label>
    <input
      name="rating"
      onClick={[Function]}
      type="radio"
      value={5}
    />
    <img
      alt="star"
      className="rating-stars__star"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="icons/Star.svg"
    />
  </label>
</div>
`;
